### DISCUSSION TITLE
Blogpost 5

### DISCUSSION THREAD

**Discussion by Hpixpoke** [LeonRother/SE_Platzverwaltung#6] [General]
Hallo zusammen,

für diese Woche stand eine kleine Demo an, mit der wir zeigen sollen, dass wir unsere gewählten Tools auch verwenden können. Wir haben im Backend zwei GET Endpoints für Gebäude und Stockwerke angelegt. Im Frontend kann man in der ersten Combobox ein Gebäude auswählen, die Liste besteht aus der API Response. Die Tabelle wird mit einem zweiten API call befüllt.

![image](https://github.com/user-attachments/assets/bea0df28-b278-404f-9ee0-d14f3020b75d)
![image](https://github.com/user-attachments/assets/bcda74c1-a3aa-468f-a63b-d994fc49ae48)

Herausfordernd war, sich erstmal durch die Dokumentation von ASP.NET und Vue.js zu forsten, weil wir beides noch nicht (nennenswert) verwendet haben. Im Backend wollten wir das SRP einhalten, wozu Singletons registriert werden mussten. Die Trennung von Controller, die die Request handeln, und Services, die die Datenzugriffe machen, war erstmal schwierig. Im Nachhinein aber eigentlich ganz simpel ;) Im Frontend haben wir uns Mühe gegeben, mit den richtigen vue directives cleanen Code zu schreiben, was uns - denke ich - auch gelungen ist. Wir stehen noch vor der Entscheidung, ob wir den Options oder Composition API style von Vue verwenden - im Code ist aktuell ersteres.

Die Datenbank haben wir so aufgesetzt, dass bei den Gebäuden ein ON DELETE CASCADE flag alle Stockwerke automatisch mit löscht. Sonst leben da noch wenige Daten - wir wollen lieber saubere Tabellendefinitionen haben, bevor wir alles hinklatschen und später neu machen müssen.

Viele Grüße
Hagen & Team Büroplatzverwaltung

**Comment by GermanJesus-lul** [comment]
Hey,
Es ist schön zu sehen, dass ihr, trotz des Einarbeitens in Vue.js und ASP.NET, noch ein gutes Ergebnis zeigen könnt. Gut finde ich vor allem, dass ihr eure Datenbank schon so weit durchdacht habt und nichts überstürzt. Es ist auch gut, dass ihr jetzt schon auf das Design eures Codes achtet.

Mit freundlichen Grüßen,
Julius und das Rezeptbuch Team
